In [24]:
import pandas as pd
import numpy as np

In [23]:
df = pd.read_csv('WELLCOME_APCspend2013_forThinkful.csv', encoding='latin_1')
df.columns

Index(['PMID/PMCID', 'Publisher', 'Journal title', 'Article title',
       'COST (£) charged to Wellcome (inc VAT when charged)'],
      dtype='object')

In [20]:
df.head()

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [102]:
# Originally I wanted to see how messy each individual title would be and turns out there were a lot of variations.
def obtain(data):
    titles = {}
    for i in data:
        print(i)
        if titles.get(i):
            titles[i] += 1
        else:
            titles[i] = 1 
    return titles 

alpha = (df['Journal title'].values)

print(obtain(alpha))

Psychological Medicine
Biomacromolecules
J Med Chem
J Med Chem
J Org Chem
Journal of Medicinal Chemistry
Journal of Proteome Research
Mol Pharm
ACS Chemical Biology
ACS Chemical Biology
Journal of Chemical Information and Modeling
Journal of Medicinal Chemistry
Journal of Medicinal Chemistry
Biochemistry
Journal of Medicinal Chemistry
Gastroenterology
Journal of Biological Chemistry
Journal of Immunology
Journal of Immunology
ACS Chemical Biology
ACS Chemical Biology
ACS Chemical Biology
ACS Chemical Neuroscience
ACS NANO
American Chemical Society
Analytical Chemistry
Analytical Chemistry
Analytical Chemistry
Bioconjugate Chemistry
Journal of Medicinal Chemistry
Journal of Medicinal Chemistry
Journal of Medicinal Chemistry 
Journal of Proteome Research
Journal of the American Chemical Society
ACS Nano
CHEST
Journal of Neurophysiology
Journal of Neurophysiology
Journal of Neurophysiology
Journal of Neurophysiology
Journal of Neurophysiology
Journal of Physiology
The Journal of Neurophys

PLoS 1
PLoS 1
PLoS 1
PLoS 1
PLoS Biology
PLoS Biology
PLoS Computational Biology
PLoS Computational Biology
PLOS COMPUTATIONAL BIOLOGY
PLOS COMPUTATIONAL BIOLOGY
PLoS Computational Biology
PLoS Computational Biology
PLoS Computational Biology
PLoS Computational Biology
PLOS Genetics
PLoS Genetics
PLoS Genetics
PLoS Genetics
PLoS Genetics
PLoS Genetics
PLoS Genetics
PLOS Genetics
PLOS Genetics
PLoS Genetics
PLoS Genetics
PLOS Genetics
PLOS Genetics
PLoS Genetics
PLoS GENETICS
PLoS Genetics
PLoS Genetics
PLOS Genetics
PLoS Genetics
PLoS Genetics
PLoS Medicine
PLoS Medicine
PLoS Medicine
PLoS Medicine
PLoS Medicine Journal
PLoS Negected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLOS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Neglected Tropical Diseases
PLoS Negle

In [41]:
# Change everything to lower case first then.
# Afterwards, get rid of the spaces to lower chances of dupes.
beta = df['Journal title']

beta = beta.str.lower()
beta = beta.str.strip()
beta = beta.apply(lambda x: str(x).replace(' ', ''))
print(beta)

0                                  psychologicalmedicine
1                                      biomacromolecules
2                                               jmedchem
3                                               jmedchem
4                                               jorgchem
5                            journalofmedicinalchemistry
6                              journalofproteomeresearch
7                                               molpharm
8                                     acschemicalbiology
9                                     acschemicalbiology
10               journalofchemicalinformationandmodeling
11                           journalofmedicinalchemistry
12                           journalofmedicinalchemistry
13                                          biochemistry
14                           journalofmedicinalchemistry
15                                      gastroenterology
16                          journalofbiologicalchemistry
17                             

In [44]:
#Get the 5 most common journals 

beta.value_counts().head(5)

plosone                         200
journalofbiologicalchemistry     53
neuroimage                       29
nucleicacidsresearch             26
plosgenetics                     24
Name: Journal title, dtype: int64

- The top 5 journals are Plosone, Journal of Biological Chemistry, Neuroimage, Nuceic Acids Research, and Plosgenetics.
- For top 5, this is usable but if we were to find more, I would have to scrub harder.

In [45]:
# Lets change the name of the cost section to type less


df['Pounds'] = df['COST (£) charged to Wellcome (inc VAT when charged)']
df.head()

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)    Pounds  
0                                              £0.00      £0.00  
1                                           £2381.04   £2381.04  
2                                            £642.56    £642.56  
3                                            £669.64    £669.64  
4                                            £685.88    £685.88

In [49]:
# Everything should be the same so lets, remove the '£' sign 

df['Pounds'] = df['Pounds'].str.replace('£', '')
# Tried turning pounds into integers earlier but saw an error so I decided to see if I could find anything. 
#Turns out there are dollar signs too, let's remove that next 
df.head(200)

PMID/PMCID  \
0                              NaN   
1                       PMC3679557   
2            23043264  PMC3506128    
3              23438330 PMC3646402   
4             23438216 PMC3601604    
5                       PMC3579457   
6                       PMC3709265   
7             23057412 PMC3495574    
8                PMCID: PMC3780468   
9                PMCID: PMC3621575   
10               PMCID: PMC3739413   
11               PMCID: PMC3530961   
12               PMCID: PMC3624797   
13                      PMC3413243   
14                      PMC3694353   
15                      PMC3572711   
16                        22610094   
17               PMCID: PMC3586974   
18    23455506  PMCID: PMC3607399    
19      PMID: 24015914 PMC3833349    
20                   : PMC3805332    
21                             NaN   
22        PMCID:\n    PMC3656742\n   
23                  PMCID: 3584654   
24             22971149 PMC3466778   
25                        23072351   
26                      PMC3477822   
27                      PMC3530960   
28            23458425 PMC3630740    
29                  PMID: 20146481   
..                             ...   
170           23885220 PMC3706957    
171  Monograph chapter, not in PMC   
172  Monograph chapter, not in PMC   
173                       23713614   
174                       23769672   
175                  PMID:23971731   
176            22738332 PMC3381227   
177                    PMC 3533981   
178                     PMC3398262   
179     PMID: 24020915 PMC3846689    
180                  PMID:23442822   
181                     PMC2843621   
182         PMCID:\n    PMC3636053   
183                        3526451   
184       PMCID:\n    PMC3664635\n   
185                        3570299   
186                     PMC3616814   
187                     PMC3566971   
188                     PMC3726342   
189                     PMC3640928   
190                     PMC3534503   
191                     PMC3579723   
192                     PMC3537519   
193                        3637822   
194       PMCID:\n    PMC3637551\n   
195       PMCID:\n    PMC3637528\n   
196                        3751660   
197         PMCID:\n    PMC3766112   
198                     PMC3528429   
199                    PMC3685541    

                                        Publisher  \
0                                             CUP   
1                                             ACS   
2                                             ACS   
3                                             ACS   
4                                             ACS   
5                                             ACS   
6                                             ACS   
7                                             ACS   
8    ACS (Amercian Chemical Society) Publications   
9    ACS (Amercian Chemical Society) Publications   
10   ACS (Amercian Chemical Society) Publications   
11   ACS (Amercian Chemical Society) Publications   
12   ACS (Amercian Chemical Society) Publications   
13                               ACS Publications   
14                               ACS Publications   
15                                  AGA Institute   
16                                          AMBSB   
17          American Association of Immunologists   
18          American Association of Immunologists   
19                      American Chemical Society   
20                      American Chemical Society   
21                      American Chemical Society   
22                      American Chemical Society   
23                      AMERICAN CHEMICAL SOCIETY   
24                      American Chemical Society   
25                      American Chemical Society   
26                      American Chemical Society   
27                      American Chemical Society   
28                      American Chemical Society   
29                      American Chemical Society   
..                      

In [62]:
# Remove the dollar signs and convert the str into int since typing .mean() earlier gave me an error 
# and told me it was a str

df['Pounds'] = df['Pounds'].str.replace('$', '')
gamma = df['Pounds']

delta = pd.to_numeric(gamma)
print(delta)

0          0.00
1       2381.04
2        642.56
3        669.64
4        685.88
5       2392.20
6       2367.95
7        649.33
8       1294.59
9       1294.78
10      1329.69
11      1287.20
12      1324.66
13       665.64
14      1006.72
15       238.08
16       265.67
17      2035.00
18      3108.08
19      1267.76
20      2286.73
21       947.07
22      1186.80
23       642.89
24      1533.29
25      2082.74
26      2314.10
27       759.67
28      1565.26
29       572.74
         ...   
2097    2236.02
2098    2275.02
2099    2373.57
2100     791.19
2101    1551.49
2102    1961.68
2103    1982.35
2104    2239.22
2105    2377.65
2106    2272.15
2107    1900.70
2108    3026.17
2109    1851.15
2110    1812.89
2111    1816.70
2112    2168.05
2113    1409.96
2114    2400.00
2115    1968.63
2116    2015.72
2117    1836.92
2118    1836.92
2119    1375.80
2120    1836.92
2121    2009.65
2122    1334.15
2123    1834.77
2124    1834.77
2125    2374.52
2126    2034.75
Name: Pounds, Length: 21

numpy.float64

In [115]:
delta.describe()


count      2127.000000
mean      24067.339972
std      146860.665559
min           0.000000
25%        1280.000000
50%        1884.010000
75%        2321.305000
max      999999.000000
Name: Pounds, dtype: float64

- The mean of COST (£) charged to Wellcome (inc VAT when charged) is 24,067.34. 
- The std is 146,860.67, and the median is 1884.01.
- There may exist an outlier in this sample that is throwing off the std and the mean.

In [143]:
# Let's try to scrub the outlier.

def neutral(data):
    result = []
    for i in data:
        if i > 10000:
            continue
        else:
            result.append(i)
    return result

n_delta = neutral(delta)

new_df = pd.DataFrame(np.array(n_delta).reshape(2077,1))

new_df.describe()

0
count  2077.000000
mean   1822.055908
std     767.698554
min       0.000000
25%    1268.700000
50%    1851.290000
75%    2302.130000
max    6000.000000

- Earlier I saw that the 75% range was 2321 so I wanted to set my new parameters as close to the upper limits as possible.
- However, this threw off my median, std, and mean.
- I didn't expect to remove so many values so I guess the truly correct numbers are still the numbers on top. 